src/material-experimental/mdc-tabs/harness/tabs-harness.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate, TestElement} from '@angular/cdk-experimental/testing';
+import {TabGroupHarnessFilters} from './tabs-harness-filters';
+
+/**
+ * Harness for interacting with a standard mat-tab-group in tests.
+ * @dynamic
+ */
+export class MatTabGroupHarness extends ComponentHarness {
+  static hostSelector = '.mat-tab-group';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a radio-button with
+   * specific attributes.
+   * @param options Options for narrowing the search
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: TabGroupHarnessFilters = {}): HarnessPredicate<MatTabGroupHarness> {
+    return new HarnessPredicate(MatTabGroupHarness);
+  }
+
+  private _tabs = this.locatorForAll(MatTabHarness);
+
+  /** Gets all tabs of the tab group. */
+  async getTabs(): Promise<MatTabHarness[]> {
+    return this._tabs();
+  }
+
+  /** Gets the selected tab of the tab group. */
+  async getSelectedTab(): Promise<MatTabHarness> {
+    for (let tab of await this.getTabs()) {
+      if (await tab.isSelected()) {
+        return tab;
+      }
+    }
+    throw new Error('No selected tab could be found.');
+  }
+}
+
+export class MatTabHarness extends ComponentHarness {
```

Comment:
Can this go in its own file?

Also needs JsDoc description

Reply:
Yes, should be possible to live in a separate file. Good point with the JSDoc description.